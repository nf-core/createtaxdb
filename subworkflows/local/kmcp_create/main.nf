include { KMCP_COMPUTE } from '../../../modules/nf-core/kmcp/compute/main'
include { KMCP_INDEX   } from '../../../modules/nf-core/kmcp/index/main'

workflow KMCP_CREATE {

    take:
    ch_fasta // channel: [ val(meta), [ fasta ] ]

    main:

    ch_versions = Channel.empty()

    KMCP_COMPUTE ( ch_fasta )
    ch_versions = ch_versions.mix(KMCP_COMPUTE.out.versions.first())

    KMCP_INDEX ( KMCP_COMPUTE.out.outdir )
    ch_versions = ch_versions.mix(KMCP_INDEX.out.versions.first())

    emit:
    db       = KMCP_INDEX.out.kmcp // channel: [ val(meta), [ db ] ]
    versions = ch_versions         // channel: [ versions.yml ]
}


