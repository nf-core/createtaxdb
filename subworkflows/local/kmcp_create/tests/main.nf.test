nextflow_workflow {

    name "Test Subworkflow KMCP_CREATE"
    script "../main.nf"
    config "./nextflow.config"
    workflow "KMCP_CREATE"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/kmcp_create"
    tag "gunzip"
    tag "kmcp"
    tag "kmcp/compute"
    tag "kmcp/index"

    test("metagenome - kmcp - fasta") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                        [[id:'db'], file(params.modules_testdata_base_path + "/genomics/prokaryotes/metagenome/fasta/sarscov2.fasta", checkIfExists: true)]).groupTuple().view()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert workflow.out.db.get(0).get(1) ==~ ".*/db-kmcp-index" },
                { assert path("${workflow.out.db[0][1]}/R001/").exists() },
                { assert snapshot(workflow.out.versions).match() }
            )
        }
    }
}
