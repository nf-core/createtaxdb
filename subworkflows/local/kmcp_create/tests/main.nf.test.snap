{
    "metagenome - kmcp - fasta": {
        "content": [
            [
                "versions.yml:md5,798729fd2552bc426f858f6dfbabbd41",
                "versions.yml:md5,ccf82c5179c57cd80a62a9ae4f975801"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-05T12:27:20.357413486"
    }
}